spectractor/extractor/spectrum.py

```diff
@@ -654,107 +749,204 @@ def load_spectrum(self, input_file_name, spectrogram_file_name_override=None,
-                        _, self.data_order2, self.err_order2 = hdu_list["ORDER2"].data
-                        if len(hdu_list) > 3:
-                            self.target.image = hdu_list["ORDER0"].data
-                            self.target.image_x0 = float(hdu_list["ORDER0"].header["IM_X0"])
-                            self.target.image_y0 = float(hdu_list["ORDER0"].header["IM_Y0"])
-                # load spectrogram info
-                if len(hdu_list) > 4:
-                    self.spectrogram = hdu_list["S_DATA"].data
-                    self.spectrogram_err = hdu_list["S_ERR"].data
-                    self.spectrogram_bgd = hdu_list["S_BGD"].data
-                    if len(hdu_list) > 7:
-                        self.spectrogram_bgd_rms = hdu_list["S_BGD_ER"].data
-                        self.spectrogram_fit = hdu_list["S_FIT"].data
-                        self.spectrogram_residuals = hdu_list["S_RES"].data
-                elif os.path.isfile(spectrogram_file_name):
-                    self.my_logger.info(f'\n\tLoading spectrogram from {spectrogram_file_name}...')
-                    self.load_spectrogram(spectrogram_file_name)
-                else:
-                    raise FileNotFoundError(f"No spectrogram info in {input_file_name} "
-                                            f"and not even a spectrogram file {spectrogram_file_name}.")
-                if "PSF_TAB" in hdu_list:
-                    self.chromatic_psf.init_table(Table.read(hdu_list["PSF_TAB"]),
-                                                  saturation=self.spectrogram_saturation)
-                elif os.path.isfile(psf_file_name):  # retro-compatibility
-                    self.my_logger.info(f'\n\tLoading PSF from {psf_file_name}...')
-                    self.load_chromatic_psf(psf_file_name)
+    def load_spectrum_latest(self, input_file_name):
+        """Load the spectrum from a fits file (data, error and wavelengths).
+
+        Parameters
+        ----------
+        input_file_name: str
+            Path to the input fits file
+
+        Examples
+        --------
+        >>> s = Spectrum(config="")
+        >>> s.load_spectrum('tests/data/reduc_20170530_134_spectrum.fits')
+        >>> print(s.units)
+        erg/s/cm$^2$/nm
+
+        .. doctest::
+            :hide:
+
+            >>> assert parameters.OBS_CAMERA_ROTATION == s.header["CAM_ROT"]
+            >>> assert parameters.CCD_REBIN == s.header["REBIN"]
+            >>> assert parameters.OBS_OBJECT_TYPE == "STAR"
+            >>> assert s.parallactic_angle == s.header["PARANGLE"]
+
+        """
+        self.header, raw_data = load_fits(input_file_name)
+        self.lambdas = raw_data[0]
+        self.lambdas_binwidths = np.gradient(self.lambdas)
+        self.data = raw_data[1]
+        if len(raw_data) > 2:
+            self.err = raw_data[2]
+            self.cov_matrix = np.diag(self.err ** 2)
+
+        # set the config parameters first
+        if self.config == "":
```

Comment:
So if someone _does_ supply a config file on init, it's used, and the stuff that's saved into the headers is just silently ignored, _and_ there's not even any way of knowing from the object where the values came from?! 😬 This seems like a recipe for sadness to me.

Reply:
I don't really think that saving where they came from is the answer though, I think that if you're loading something pre-made and the config is in the headers then it should be illegal to provide a config file.

Reply:
That's a very good remark. Hmmm then I add a check at the beginning of the generic `Spectrum.load_spectrum()` function (if older than 2.4 need a config file, otherwise do not give one) and remove the if in this `load_spectrum_latest()` function.

Reply:
Doing that I detected that it is not safe to save the parameters like DEBUG, VERBOSE, SAVE, PLOT_DIR in the Spectrum header, because then Spectractor run with this behaviour which may be not the one taht the user wants. So I created a STYLE_PARAMETERS list which contains the parameters that we don't want to save in the output header (and which will not be loaded then).

Reply:
Oh yes, perfect. Certainly behaviour-type parameters like that vs science config parameters are an excellent thing to separate!